llvm/include/llvm/Analysis/AliasAnalysis.h

```diff
@@ -381,6 +381,14 @@ class AAResults {
                      MemoryLocation::getBeforeOrAfter(V2));
   }
 
+  /// A convenience wrapper around the \c isNoAlias for two instructions.
+  bool isNoAlias(const Instruction *I1, const Instruction *I2) {
```

Comment:
Using NoAlias terminology isn't right. Aliasing is a concept of memory locations. For instructions, you are interested in memory effects, as provided by getModRefInfo(). Adding a getModRefInfo() overload that accepts two instructions would be fine.

Reply:
I'd expect the implementation to look somewhat similar to getModRefInfo() for an instruction and call: https://github.com/llvm/llvm-project/blob/d254fa877f419e61e54709f0a6f2e891da893a60/llvm/lib/Analysis/AliasAnalysis.cpp#L190-L209